pyiron_base/storage/parameters.py

```diff
@@ -622,6 +622,8 @@ def __setitem__(self, key, value):
             key (str): key to be set of modified
             value (float/int/str): value to be set
         """
+        if isinstance(value, np.ndarray):
+            value = value.tolist()
         if isinstance(key, int):
             if self.read_only and self._dataset["Value"][key] != value:
                 self._read_only_error()
```

Comment:
Currently, i.e. without this modification, the value would be just assigned to the correct key - I do not see type checking or the like here. If the new lines are not in, the array is stored as is?

Reply:
I don't know why, but I see that it becomes `str`. There are SOOOOOOOOOOOO many problems with `GenericParameters` that I don't understand. So don't ask me why...

Reply:
is it the same if you assign something that has a higher dimension? For example `np.ones((3, 3))`?

Reply:
Yes... The full MWE is: 
```
from pyiron_base import Project, GenericParameters
import numpy as np
pr = Project('.')
hdf = pr.create_hdf('some_test_hdf', 'genericParameter')
a = GenericParameters(table_name='gp')
a['some']= np.ones((3, 3))
a.to_hdf(hdf)
a.from_hdf(hdf)
a['some']
```
returning the plain `np.ndarray`?!

Therefore, the `str` conversion has to be caused by something else?!

Reply:
ok I'm more than confused now. Indeed the code you pasted gives `ndarray` **if** they are executed in one cell. If I add the following lines in a **different** cell and execute them, I get `str`.

```python
a['some']= np.ones((3, 3))
a.to_hdf(hdf)
a.from_hdf(hdf)
a['some']
```

Reply:
Ah no sorry, it has nothing to do with different cells but it's about doing the same thing twice, i.e.

```python
from pyiron_base import GenericParameters
gp = GenericParameters(table_name='test')
gp['test'] = np.ones((3, 3))
gp['test'] = np.ones((3, 3))
type(gp['test'])
```
gives `str` while

```python
from pyiron_base import GenericParameters
gp = GenericParameters(table_name='test')
gp['test'] = np.ones((3, 3))
type(gp['test'])
```

gives `ndarray`.

Reply:
Funny enough... I see you found the same line in the code causing this behavior... The question now is: Why was this converted to `str` in the first place 😢